addons/sale_timesheet/tests/test_sale_service.py

```diff
@@ -656,3 +656,68 @@ def test_remaining_hours_prepaid_services(self):
         })
         self.assertEqual(timesheet.so_line, prepaid_service_sol, "The SOL should be the same than one containing the prepaid service product.")
         self.assertEqual(prepaid_service_sol.remaining_hours, 2, "The remaining hours should not change.")
+
+    def test_several_uom_sol_to_planned_hours(self):
+        planned_hours_for_uom = {
+            'day': 8.0,
+            'hour': 1.0,
+            'unit': 1.0,
+            'gram': 0.0,
+        }
+        
+        Product = self.env['product.product']
+        product_vals = {
+            'type': 'service',
+            'service_type': 'timesheet',
+            'project_id': self.project_global.id,
+            'service_tracking': 'task_global_project',
+        }
+
+        SaleOrderLine = self.env['sale.order.line']
+        sol_vals = {
+            'product_uom_qty': 1,
+            'price_unit': 100,
+            'order_id': self.sale_order.id,
+        }
+
+        self.project_global.task_ids = False
+        for uom_name in list(planned_hours_for_uom):
```

Comment:
I think it is equivalent to:
```suggestion
        for uom_name in planned_hours_for_uom:
```
🤔 

If not then it is equivalent to:
```suggestion
        for uom_name in planned_hours_for_uom.keys():
```